content/client-lib-development-guide/features.textile

```diff
@@ -1331,7 +1331,7 @@ class ClientOptions:
   maxFrameSize: Int default 524288 // TO3l8
 
 class AuthOptions: // RSA8e
-  authCallback: (() -> io (String | TokenDetails | TokenRequest))? // RSA4a, RSA4, TO3j5, AO2b
+  authCallback: (TokenParams, () -> io (String | TokenDetails | TokenRequest))? // RSA4a, RSA4, TO3j5, AO2b
```

Comment:
(Hi! Just visiting :) )

This should actually be `((TokenParams) -> io ...)?`. As you wrote it, it would be an (optional) tuple of `TokenParams` and a function that takes no arguments and returns one of `String | TokenDetails | TokenRequest`. You want `TokenParams` to be an argument of that function instead (if my Ably-fu is not too rusty...).

Reply:
@tcard oops, thanks. Hope things are going well for you.